src/LogFileReader.cpp

```diff
@@ -67,19 +72,27 @@ bool LogFileReader::readAll(const QString& fileName)
             emit batteryDataReceived(batteryData);
         }
     }
-
     return true;
 }
 
 // File input is a csv file in the format of hh:mm:ss:zzz, voltage, current
 bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) const
 {
-    // TODO implement this first
-
-    // This is here to the compiler happy. Otherwise the compile
-    // will have an error warning about an unused variable. Remove this
-    // when you use it.
-    Q_UNUSED(line);
-    Q_UNUSED(batteryData);
-    return true;
+    QStringList lines = line.split(COMMA);
+    QTime time = QTime::fromString(lines[0], TIMEFORMAT);
+    bool notok = false;
+    if (time.isValid() && lines.count() == 3 && (lines[1].toDouble(&notok) != notok) && (lines[2].toDouble(&notok) != notok))
```

Comment:
Hmm, not quite what I was intending. As a style thing you want to avoid very long lines, doing 
```c++
if (time.isValid() && lines.count() == 3 && 
    lines[1].toDouble(&notok) != notok && 
    lines[2].toDouble(&notok) != notok)
```
Would look better.

But more importantly, I want you to think about
`lines[1].toDouble(&notok) != notok` and why it won't work.